src/backend/wsl.ts

```diff
@@ -1306,15 +1306,16 @@ export default class WSLBackend extends events.EventEmitter implements VMBackend
     this.currentAction = Action.STOPPING;
     try {
       await this.setState(State.STOPPING);
-      await this.vtun.stop();
       await this.kubeBackend.stop();
 
       await this.progressTracker.action('Shutting Down...', 10, async() => {
         if (await this.isDistroRegistered({ runningOnly: true })) {
           await this.execCommand('/usr/local/bin/wsl-service', '--ifstarted', 'k3s', 'stop');
           await this.execCommand('/usr/local/bin/wsl-service', '--ifstarted', 'docker', 'stop');
           await this.execCommand('/usr/local/bin/wsl-service', '--ifstarted', 'containerd', 'stop');
+          await this.execCommand('/usr/local/bin/wsl-service', '--ifstarted', 'rancher-desktop-guestagent', 'stop');
           await this.execCommand('/usr/local/bin/wsl-service', '--ifstarted', 'buildkitd', 'stop');
+          await this.vtun.stop();
```

Comment:
`this.vtun.stop()` should happen even if the distro is not running (but probably after we tried to stop the guest agent).

So, move it to next to `this.process?.kill()`.